### PR TITLE
Work branch: Auth module

### PR DESCRIPTION
**Pull request by erastian**
- Changed .env CLIENT_URL var. Added http:// prefix before url for unifying use.
- In auth added forgot-password and restore-password methods
- For forgot-password and restore-password methods was added related methods in auth.service, users.service and users.repository
- Changed related interfaces
- Added template and method in mailer service for sending password restoration link
- In auth.controller fixed misspelled name of users.service
- Minor but useful fixes for mailer & auth module 
- Shifted code to try/catch block

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -86,16 +86,34 @@ export class AuthController extends BaseController implements IAuthController {
...
+		}
 	}
 
 	async restorePassword(req: Request, res: Response, next: NextFunction): Promise<void> {
-		this.ok(res, {});
+		const user = req.user;
+		const { newPassword } = req.body;
+
+		try {
+			await this.authService.restorePassword(user, newPassword);
```
Comment (mdziarkach) [inline src/auth/auth.controller.ts]: please provide only the necessary information for the service
```js
await this.authService.restorePassword(user.id, newPassword);
```

File: src/mailer/mailer.service.ts [hunk 2]
```diff
@@ -60,4 +60,28 @@ export class MailerService implements IMailerService {
...
+			from: `Happy Admin <${this.configService.get('SMTP_USER')}>`,
+			to: `${capitalizeUsername} <${emailFor}>`,
+			subject: '[Restore password] Password restoration message',
+			template: 'restore_password',
+			context: {
+				username: capitalizeUsername,
+				server_name: SERVER_NAME,
+				link: link,
+			},
+		};
```
Comment (mdziarkach) [inline src/mailer/mailer.service.ts]: move please code above inside of `try` section

Comment (erastian) [reply src/mailer/mailer.service.ts]: Done

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (erastian) [reply src/auth/auth.controller.ts]: Done

File: src/auth/auth.controller.ts [hunk 3]
```diff
@@ -86,16 +86,34 @@ export class AuthController extends BaseController implements IAuthController {
...
+		const user = req.user;
+		const { newPassword } = req.body;
+
+		try {
+			await this.authService.restorePassword(user.id, newPassword);
+
+			this.ok(res, 'Password was successfully changed');
+		} catch (e) {
+			return next(e);
+		}
```
Comment (mdziarkach) [inline src/auth/auth.controller.ts:111]:
```suggestion
try {
  const user = req.user;
  const { newPassword } = req.body;
  await this.authService.restorePassword(user.id, newPassword);

  this.ok(res, 'Password was successfully changed');
} catch (e) {
  return next(e);
}
```

Comment (erastian) [reply src/auth/auth.controller.ts:111]: Done

**Pull request merged**